### ISSUE TITLE
Track macronutrients beyond protein

### ISSUE THREAD

**Issue by paulzakin** [successible/cleanslate#15]
Likely total fat and total carbohydrates.

We would put it behind a feature flag to keep the interface from being too cluttered.

**Comment by paulzakin** [comment]
We will reopen this request if we get many other users requesting it

**Comment by kimjongbing** [comment]
> We will reopen this request if we get many other users requesting it

it would be cool to add this since i track my fat, carbohydrates, fibre as well

**Comment by paulzakin** [comment]
After thinking about this for the past month or two, I have decided I will not be implementing this feature for two reasons.

1. Macronutrients are a slippery slope. First, people want total carbohydrates and total fat. Then they want fat, broken down by saturated and monounsaturated fat. Then they want carbohydrates split by fiber and sugar alcohols. Then, they want vitamins. Then they want minerals. And to be clear, it is fine for people to want these things! However, it opens a Pandora’s box of endless complexity. As a one-man development team working on this app in their free time, that scares me. Not to mention there are already many apps where you can track those things.

2. Clean Slate is an app with a pretty opinionated philosophy. One, we believe existing apps are too complex and “numbers-oriented”, especially for those who struggle with black-and-white thinking. Hence, it is vital to minimize as much complexity and numbers to track as possible. Two, we believe that the literature shows that the only macronutrients worth tracking for the person trying to lose weight are calories and protein. Calories because weight loss ultimately is just calories in and calories out. Protein is the primary driver of satiety muscle retention and growth. The rest of the macronutrients might make a difference around the margin, but we view them as mostly noise. In fact, we view them as actively harmful noise. This is for three reasons. One, they give people additional targets that they can "fail". Two, they can lock people into restrictive diet plans. Three, they makes the interface of app, much more cluttered and complex.

In short, anyone who wants this feature will need to use a different app or fork Clean Slate, which they are welcome to do, as it is open source 😊

**Comment by paulzakin** [comment]
And thank you to everyone who asked for this feature for understanding this decision. I really appreciate it!
